packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -25,41 +28,65 @@ const defaultProps = {
   isHovered: false,
   component: 'a',
   isDisabled: false,
-  href: '#'
+  href: '#',
+  onSelect: Function.prototype,
+  onClick: Function.prototype
 };
 
-const DropdownItem = ({ className, children, isHovered, component: Component, isDisabled, ...props }) => {
-  const additionalProps = props;
-  if (Component === 'a') {
-    additionalProps['aria-disabled'] = isDisabled;
-    additionalProps.tabIndex = isDisabled ? -1 : additionalProps.tabIndex;
-  } else if (Component === 'button') {
-    additionalProps.disabled = isDisabled;
-  }
+class DropdownItem extends React.Component {
+  render() {
+    const {
+      className,
+      children,
+      isHovered,
+      onClick,
+      component: Component,
+      isDisabled,
+      ...additionalProps
+    } = this.props;
+    if (Component === 'a') {
+      additionalProps['aria-disabled'] = isDisabled;
+      additionalProps.tabIndex = isDisabled ? -1 : additionalProps.tabIndex;
+    } else if (Component === 'button') {
+      additionalProps.disabled = isDisabled;
+    }
 
-  return (
-    <li>
-      {React.isValidElement(children) ? (
-        React.Children.map(children, child =>
-          React.cloneElement(child, {
-            className: `${css(
-              isDisabled && styles.modifiers.disabled,
-              isHovered && styles.modifiers.hover,
-              className
-            )} ${child.props.className}`
-          })
-        )
-      ) : (
-        <Component
-          {...additionalProps}
-          className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
-        >
-          {children}
-        </Component>
-      )}
-    </li>
-  );
-};
+    return (
+      <DropdownContext.Consumer>
+        {onSelect => (
+          <li>
+            {React.isValidElement(children) ? (
+              React.Children.map(children, child =>
+                React.cloneElement(child, {
+                  className: `${css(
+                    isDisabled && styles.modifiers.disabled,
+                    isHovered && styles.modifiers.hover,
+                    className
+                  )} ${child.props.className}`
+                })
+              )
+            ) : (
+              <Component
+                {...additionalProps}
+                className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
+                onClick={event => {
+                  if (!isDisabled) {
+                    if (Component === 'button') onClick && onClick(event);
+                    onSelect && onSelect(event);
+                  } else {
+                    Function.prototype;
```

Comment:
Getting an eslint error here

> Expected an assignment or function call and instead saw an expression.eslint(no-unused-expressions)

I think, if we want the handler to do nothing in case `isDisabled` is true, we can just omit this `else` block